aws_lambda_powertools/event_handler/util.py

```diff
@@ -40,3 +69,30 @@ def extract_origin_header(resolver_headers: Dict[str, Any]):
         return resolved_header[0]
 
     return resolved_header
+
+
+def validate_openapi_security_parameters(
+    security: List[Dict[str, List[str]]],
+    security_schemes: Optional[Dict[str, "SecurityScheme"]],
+) -> bool:
+    """
+    Validates the security parameters based on the provided security schemes.
+
+    Parameters
+    ----------
+    security: List[Dict[str, List[str]]]
+        A list of security requirements
+    security_schemes: Optional[Dict[str, "SecurityScheme"]]
+        A dictionary mapping security scheme names to their corresponding security scheme objects.
+
+    Returns
+    -------
+    bool
+        True if all security scheme names in the `security` parameter are present in the `security_schemes` parameter,
+        False otherwise.
+
+    """
+
+    return bool(
+        security_schemes and all(key in security_schemes for sec in security for key in sec),
+    )
```

Comment:
>Always guard against a None. Otherwise the all will raise TypeError: argument of type 'NoneType' is not iterable

In this case will not because the first expression test if security_schemes is defined. But your code makes it more clear.